drivers/hmis/app/models/hmis/hud/processors/custom_assessment_processor.rb

```diff
@@ -26,5 +22,48 @@ def schema
 
     def information_date(_)
     end
+
+    def post_process
+      validate_alt_aha_score
+    end
+
+    private
+
+    # Validate Alt-AHA score if present. For more information on Alt-AHA score see https://docs.google.com/document/d/1Gcz9-t_utRcqGV9xCzQvTehjQOCqqPv_5-JY_IhhL4Q/edit?tab=t.0#heading=h.kpe3ch74jsjj
+    # If assessment contained an Alt-AHA score calculation, validate it by re-calculating the score.
+    # If the scores don't match, return a validation error.
+    # This prevents users from submitting scores that do not match the current form values.
+    def validate_alt_aha_score
+      alt_aha_item = @processor.definition.link_id_item_hash.values.find { |i| i.component == 'ALT_AHA' }
+      return unless alt_aha_item # alt-aha not collected on this form, nothing to validate
+
+      alt_aha_custom_field_key = alt_aha_item.dig('mapping', 'custom_field_key')
+      raise 'Alt-AHA form item does not map to a custom field' unless alt_aha_custom_field_key # unexpected
+
+      value = @hud_values.fetch(alt_aha_custom_field_key)
+      return unless value # Skip if there's no score to validate
+
```

Comment:
We should probably add this as well, if value is submitted as "hidden" it would still be present in hud_values obj
```suggestion
return if value == HIDDEN_FIELD_VALUE

```